x-pack/plugin/slm/src/main/java/org/elasticsearch/xpack/slm/action/TransportPutSnapshotLifecycleAction.java

```diff
@@ -144,18 +145,26 @@ public ClusterState execute(ClusterState currentState) {
             } else {
                 Map<String, SnapshotLifecyclePolicyMetadata> snapLifecycles = new HashMap<>(snapMeta.getSnapshotConfigurations());
                 SnapshotLifecyclePolicyMetadata oldLifecycle = snapLifecycles.get(id);
-                SnapshotLifecyclePolicyMetadata newLifecycle = SnapshotLifecyclePolicyMetadata.builder(oldLifecycle)
-                    .setPolicy(request.getLifecycle())
-                    .setHeaders(filteredHeaders)
-                    .setVersion(oldLifecycle == null ? 1L : oldLifecycle.getVersion() + 1)
-                    .setModifiedDate(Instant.now().toEpochMilli())
-                    .build();
-                snapLifecycles.put(id, newLifecycle);
-                lifecycleMetadata = new SnapshotLifecycleMetadata(snapLifecycles, currentMode, snapMeta.getStats());
-                if (oldLifecycle == null) {
-                    logger.info("adding new snapshot lifecycle [{}]", id);
+
+                if (oldLifecycle != null
```

Comment:
I don't think the `oldLifecycle` can be `null` here.

Reply:
It could be null here if it's  a new policy ... but I agree, better to do it like `isNoopUpdate`

Reply:
Wouldn't it fall into the block preceding the `} else {` then?

Reply:
It's not a super important point either way -- even the `isNoopUpdate` implementation that I'd like you to emulate seems a bit squishy on it (or so it looks to me at a first glance).